copy/copy.go

```diff
@@ -569,20 +584,29 @@ func (ic *imageCopier) copyUpdatedConfigAndManifest(ctx context.Context) ([]byte
 		}
 		pendingImage = pi
 	}
-	manifest, _, err := pendingImage.Manifest(ctx)
+	manifestBytes, _, err := pendingImage.Manifest(ctx)
 	if err != nil {
 		return nil, errors.Wrap(err, "Error reading manifest")
 	}
+	if expectedDigest != nil {
+		manifestDigest, err := manifest.Digest(manifestBytes)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Error computing digest of pending manifest")
+		}
+		if manifestDigest != *expectedDigest {
```

Comment:
(in addition to moving the code, as discussed elsewhere)

Please use `manifest.MatchesDigest`, to have some chance of supporting multiple different digest algorithms in the future.

Reply:
Ah, right.  Changing it.